workers/data_refinery_workers/processors/salmon.py

```diff
@@ -0,0 +1,258 @@
+from __future__ import absolute_import, unicode_literals
+import os
+import gzip
+import tarfile
+from typing import Dict
+from celery import shared_task
+from data_refinery_workers.processors import utils
+from data_refinery_common.job_lookup import Downloaders
+from data_refinery_common.models import Batch, BatchStatuses, BatchKeyValue
+from data_refinery_common.logging import get_and_configure_logger
+import subprocess
+
+
+logger = get_and_configure_logger(__name__)
+JOB_DIR_PREFIX = "processor_job_"
+
+
+def _set_job_prefix(job_context: Dict) -> str:
+    job_context["job_dir_prefix"] = JOB_DIR_PREFIX + str(job_context["job_id"])
+    return job_context
+
+
+def _prepare_files(job_context: Dict) -> Dict:
+    """Moves the file(s) from the raw directory to the temp directory.
+
+    Also adds the keys "input_file_path" and "output_directory" to
+    job_context so everything is prepared for processing. If the reads
+    are paired then there will also be an "input_file_path_2" key
+    added to job_context for the second read.
+    """
+    # Salmon processor jobs have only one batch per job, but may have
+    # up to two files per batch.
+    batch = job_context["batches"][0]
+    files = batch.files
+
+    for file in files:
+        try:
+            file.download_raw_file(job_context["job_dir_prefix"])
+        except Exception:
+            logger.exception("Exception caught while retrieving raw file %s",
+                             file.get_raw_path(),
+                             processor_job=job_context["job_id"],
+                             batch=batch.id)
+
+            failure_template = "Exception caught while retrieving raw file {}"
+            job_context["job"].failure_reason = failure_template.format(file.name)
+            job_context["success"] = False
+            return job_context
+
+    job_context["input_file_path"] = files[0].get_temp_pre_path(job_context["job_dir_prefix"])
+    # Salmon outputs an entire directory of files, so create a temp
+    # directory to output it to until we can zip it to
+    # files[0].get_temp_post_path()
+    job_context["output_directory"] = os.path.join(
+        files[0].get_temp_dir(job_context["job_dir_prefix"]), "output")
+    os.makedirs(job_context["output_directory"], exist_ok=True)
+
+    if len(files) == 2:
```

Comment:
Do we ever expect the the length of files to be something other than 0, 1, or 2? Should we handle these cases?

Reply:
There should always be exactly one or two files for salmon jobs, but I will add a check to make sure this is the case.